### PR TITLE
Added support for helm deployment of kubefin

### PR DESCRIPTION
**Pull request by czhfe**
1. added support for helm deployment of kubefin.
2. added support for prometheus scraping kubefin-agent metrics, which can replace Opentelemetry Collector.

### CODE REVIEW

**Comment by mrhello369** [comment]
Nice work!
Thanks for your contribution, I will review it asap.

File: charts/kubefin-agent/Chart.yaml [hunk 1]
```diff
@@ -0,0 +1,12 @@
+apiVersion: v2
+name: kubefin-agent
+description: A Helm chart for Kubefin-agent
```
Comment (mrhello369) [inline charts/kubefin-agent/Chart.yaml:3]: typo: kubefin-agent

File: charts/kubefin-agent/Chart.yaml [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+sources:
+  - https://github.com/kubefin/kubefin
+keywords:
+  - kubefin
+  - kubefin-agent
+  - FinOps
+type: application
+version: 0.1.0
+appVersion: "0.1.2"
```
Comment (mrhello369) [inline charts/kubefin-agent/Chart.yaml:12]: Do we need to change it manually for each different release? 
If yes, can we make it automatically?

File: charts/kubefin-agent/templates/kubefin-agent-podmonitor.yaml [hunk 3]
```diff
@@ -0,0 +1,37 @@
+{{- if .Values.podMonitor.enabled }}
+apiVersion: monitoring.coreos.com/v1
+kind: PodMonitor
```
Comment (mrhello369) [inline charts/kubefin-agent/templates/kubefin-agent-podmonitor.yaml:3]: What is PodMonitor? What's the purpose to install it?

File: charts/kubefin-agent/values.yaml [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+# - --v=6
+
+config: |
+  CLOUD_PROVIDER: "default"
+  CLUSTER_NAME: "kubefin-server"
+  CLUSTER_ID: ""
+  CUSTOM_CPU_CORE_HOUR_PRICE: ""
+  CUSTOM_RAM_GB_HOUR_PRICE: ""
+  CPU_CORE_RESERVED: "0.1"
+  NODE_RAM_RESERVED: "0.1"
```
Comment (mrhello369) [inline charts/kubefin-agent/values.yaml:20]: Please change to:
````
NODE_CPU_DEVIATION: "0.0"
NODE_RAM_DEVIATION: "0.4"
```
In real situation(on-premise), the real memory we can't get from node's yaml, it has some difference. We use this to make it more accurate.

File: charts/kubefin/.helmignore [hunk 5]
```diff
@@ -0,0 +1,23 @@
+# Patterns to ignore when building packages.
+# This supports shell glob matching, relative path matching, and
+# negation (prefixed with !). Only one pattern per line.
+.DS_Store
+# Common VCS dirs
```
Comment (mrhello369) [inline charts/kubefin/.helmignore:5]: This file is autogenerated?

File: charts/kubefin-agent/templates/NOTES.txt
Comment (mrhello369) [inline charts/kubefin-agent/templates/NOTES.txt:1]: This file is empty, can we delete it?

**Review by mrhello369 — COMMENTED** [review]
Nice works, just few nits

File: charts/kubefin-agent/Chart.yaml (hunk 1 above)
Comment (czhfe) [reply charts/kubefin-agent/Chart.yaml:3]: > typo: kubefin-agent

I didn't see where the mistake was

File: charts/kubefin-agent/Chart.yaml (hunk 2 above)
Comment (czhfe) [reply charts/kubefin-agent/Chart.yaml:12]: > Do we need to change it manually for each different release? If yes, can we make it automatically?

We don't necessarily need every version to modify it.

File: charts/kubefin-agent/templates/kubefin-agent-podmonitor.yaml (hunk 3 above)
Comment (czhfe) [reply charts/kubefin-agent/templates/kubefin-agent-podmonitor.yaml:3]: > 

PodMonitor is a custom resource for PrometheusOperator, using which you can integrate with existing Prometheus.Metrics collection tasks can be added to Prometheus through PrometheusOperator's PodMonitor resource.This is an option that can be used in place of the Opentelemetry Collector, reusing existing Prometheus infrastructure.

File: charts/kubefin/.helmignore (hunk 5 above)
Comment (czhfe) [reply charts/kubefin/.helmignore:5]: > This file is autogenerated?

Yes, it allows helm to ignore files that we need to ignore, similar to .gitignore.

File: charts/kubefin-agent/templates/NOTES.txt
Comment (czhfe) [reply charts/kubefin-agent/templates/NOTES.txt:1]: > This file is empty, can we delete it?

NOTES.txt contains a brief description of the chart, if the content of the brief description is not needed we can delete it.

File: charts/kubefin-cost-analyzer/values.yaml [hunk 6]
```diff
@@ -0,0 +1,125 @@
...
+
+dashboard:
+  replicaCount: 1
+  image:
+    repository: kubefin/kubefin-dashboard
+    pullPolicy: IfNotPresent
+    tag: "v0.1.2"
+
+  resources: {}
+  #  requests:
```
Comment (mrhello369) [inline charts/kubefin-cost-analyzer/values.yaml:67]: Why we comment this?

**Review by mrhello369 — COMMENTED** [review]
Other lgtm

File: charts/kubefin-agent/Chart.yaml (hunk 2 above)
Comment (mrhello369) [reply charts/kubefin-agent/Chart.yaml:12]: Can u help create a issue to follow this problem. Thanks

File: charts/kubefin-cost-analyzer/values.yaml (hunk 6 above)
Comment (czhfe) [reply charts/kubefin-cost-analyzer/values.yaml:67]: > Why we comment this?

It is often recommended not to specify a default resources, but to let the user choose it. It is up to the user to choose. This allows the chart to run in less resource-intensive environments, but if you do want to specify a resource, uncomment the line under "resource:" and remove the curly brackets after it, adjusting as necessary.

**Review by mrhello369 — APPROVED** [review]
lgtm

**Pull request merged**